File: app/routers/tts.py
# OneFlow TTS 接口 — edge-tts（免费无 Key）
from fastapi import APIRouter, Depends, HTTPException
from fastapi.responses import Response
from pydantic import BaseModel

from ..deps import get_current_user
from ..models import User

router = APIRouter(prefix="/api/tts", tags=["tts"])

MAX_TEXT_LENGTH = 2000


class TtsRequest(BaseModel):
    text: str
    voice: str = "zh-CN-XiaoxiaoNeural"


@router.post("")
async def synthesize_tts(req: TtsRequest, user: User = Depends(get_current_user)) -> Response:
    text = req.text.strip()
    if not text:
        raise HTTPException(status_code=400, detail="文本不能为空")
    if len(text) > MAX_TEXT_LENGTH:
        raise HTTPException(status_code=400, detail="文本过长")

    try:
        import edge_tts

        communicate = edge_tts.Communicate(text, req.voice)
        audio_bytes = b""
        async for chunk in communicate.stream():
            if chunk.get("type") == "audio":
                audio_bytes += chunk["data"]
    except Exception as e:
        raise HTTPException(status_code=502, detail=f"语音合成失败: {e}")

    return Response(content=audio_bytes, media_type="audio/mpeg")

File: frontend/src/api/tts.ts
import { http } from './http'

export async function tts(text: string, voice?: string): Promise<Blob> {
  try {
    const resp = await http.post(
      '/tts',
      { text, voice: voice ?? 'zh-CN-XiaoxiaoNeural' },
      // 8 秒超时：edge-tts 走微软服务器国内偶发慢请求，超时即回退浏览器语音，
      // 绝不让播报链路挂住几十秒
      { responseType: 'blob', timeout: 8000 },
    )
    return resp.data as Blob
  } catch {
    // blob 错误响应时 error.response.data 可能是 Blob 而非对象，无法读取 detail，统一抛固定文案
    throw new Error('语音合成失败')
  }
}
